scalyr_agent/builtin_monitors/linux_process_metrics.py

```diff
@@ -638,7 +639,18 @@ def run_single_cycle(self, collector=None):
         @rtype:
         """
 
-        num_fds = len(os.listdir(self.__path))
+        num_fds = 0
+        try:
+            num_fds = len(os.listdir(self.__path))
+        except OSError, e:
+            if e.errno == errno.ENOENT:
+                # ignore file not found, it just means the process
+                # is dead so log a warning, but still continue
+                self._logger.warn( "Path '%s' not found.  This likely means that process '%ld' is no longer running" % (self.__path, self.__pid ) )
```

Comment:
Let's not even bother to log the warning here.. this is kind of expect behavior.. a process may exit that is being monitored.  

I think it's ok to consider it having zero fds if the process has exited.